bootcd/dmsquash-live-root.toram.sh

```diff
@@ -0,0 +1,241 @@
+           sleep 5
+        fi
+ 
+        dd if=/dev/null of=/overlay bs=1024 count=1 seek=$((512*1024)) 2> /dev/null
+        losetup $OVERLAY_LOOPDEV /overlay
+    fi
+ 
+    # set up the snapshot
+    echo 0 `blockdev --getsz $BASE_LOOPDEV` snapshot $BASE_LOOPDEV $OVERLAY_LOOPDEV p 8 | dmsetup create $readonly_overlay live-rw
+}
+ 
+# live cd helper function
+do_live_from_base_loop() {
+    do_live_overlay
+}
+ 
+# we might have a genMinInstDelta delta file for anaconda to take advantage of
+if [ -e /dev/.initramfs/live/${live_dir}/osmin.img ]; then
+    OSMINSQFS=/dev/.initramfs/live/${live_dir}/osmin.img
+fi
+ 
+if [ -n "$OSMINSQFS" ]; then
+    # decompress the delta data
+    dd if=$OSMINSQFS of=/osmin.img 2> /dev/null
+    OSMIN_SQUASHED_LOOPDEV=$( losetup -f )
+    losetup -r $OSMIN_SQUASHED_LOOPDEV /osmin.img
+    mkdir -p /squashfs.osmin
+    mount -n -t squashfs -o ro $OSMIN_SQUASHED_LOOPDEV /squashfs.osmin
+    OSMIN_LOOPDEV=$( losetup -f )
+    losetup -r $OSMIN_LOOPDEV /squashfs.osmin/osmin
+    umount -l /squashfs.osmin
+fi
+ 
+# we might have just an embedded ext3 to use as rootfs (uncompressed live)
+if [ -e /dev/.initramfs/live/${live_dir}/ext3fs.img ]; then
+  EXT3FS="/dev/.initramfs/live/${live_dir}/ext3fs.img"
+fi
+ 
+if [ -n "$EXT3FS" ] ; then
+    BASE_LOOPDEV=$( losetup -f )
+    losetup -r $BASE_LOOPDEV $EXT3FS
+ 
+    # Create overlay only if toram is not set
+    if [ -z "$toram" ] ; then
+        do_live_from_base_loop
+    fi
+fi
+ 
+# we might have an embedded ext3 on squashfs to use as rootfs (compressed live)
+if [ -e /dev/.initramfs/live/${live_dir}/squashfs.img ]; then
+  SQUASHED="/dev/.initramfs/live/${live_dir}/squashfs.img"
+fi
+ 
+if [ -e "$SQUASHED" ] ; then
+    if [ -n "$live_ram" ] ; then
+        echo "Copying live image to RAM..."
+        echo "(this may take a few minutes)"
+        dd if=$SQUASHED of=/squashed.img bs=512 2> /dev/null
+        umount -n /dev/.initramfs/live
+        echo "Done copying live image to RAM."
+        if [ ! -n "$no_eject" ]; then
+            eject -p $livedev || :
+        fi
+        SQUASHED="/squashed.img"
+    fi
+ 
+    SQUASHED_LOOPDEV=$( losetup -f )
+    losetup -r $SQUASHED_LOOPDEV $SQUASHED
+    mkdir -p /squashfs
+    mount -n -t squashfs -o ro $SQUASHED_LOOPDEV /squashfs
+ 
+    BASE_LOOPDEV=$( losetup -f )
+    losetup -r $BASE_LOOPDEV /squashfs/LiveOS/ext3fs.img
+ 
+    umount -l /squashfs
+ 
+    # Create overlay only if toram is not set
+    if [ -z "$toram" ] ; then
+        do_live_from_base_loop
+    fi
+fi
+ 
+# If the kernel parameter toram is set, create a tmpfs device and copy the 
+# filesystem to it. Continue the boot process with this tmpfs device as
+# a writable root device.
+if [ -n "$toram" ] ; then
+    blocks=$( blockdev --getsz $BASE_LOOPDEV )
+ 
+    echo "Create tmpfs ($blocks blocks) for the root filesystem..."
+    mkdir -p /image
+    mount -n -t tmpfs -o nr_blocks=$blocks tmpfs /image
```

Comment:
hmm... yea we could make it dynamic, perhaps a % of ram up to a max?
First pass though, this gets us out of the woods